flex/engines/graph_db/runtime/adhoc/operators/scan.cc

```diff
@@ -150,6 +150,81 @@ bool parse_idx_predicate(const algebra::IndexPredicate& predicate,
   return true;
 }
 
+bool parse_idx_predicate(const algebra::IndexPredicate& predicate,
+                         const std::map<std::string, std::string>& params,
+                         std::vector<Any>& oids, bool& scan_oid) {
```

Comment:
Cypher queries scan one oid at a time, while Gremlin allows specifying multiple oids in an array. We might implement support for a vector of oids, treating single id scans as a special case.